scheduler/src/cook/kubernetes/api.clj

```diff
@@ -795,7 +800,10 @@
         {:keys [custom-shell init-container set-container-cpu-limit? sidecar]} (config/kubernetes)
         checkpoint (calculate-effective-checkpointing-config job task-id)
         job-submit-time (tools/job->submit-time job)
-        image (calculate-effective-image (config/kubernetes) job-submit-time image checkpoint task-id)
+        pod-name (str task-id)
+        image (if (synthetic-pod? (or pod-name ""))
```

Comment:
When is pod-name empty? (Can a pod name in k8s be the empty string?!?)

Reply:
I don't think it should ever be. But if it is, my code would be introducing an NPE
I'm fine dropping the `or`

Reply:
Lets do that, otherwise, a future reviewer can be left guessing when this might be blank. (And may see illusionary bugs when other code isn't robust to an empty pod name.)

Reply:
updated